### ISSUE TITLE
Set cell cursor to pointer if onRowClick is set

### ISSUE THREAD

**Issue by ryanmk54** [KevinVandy/material-react-table#6]
### Describe the bug

Thank you for this awesome library!

It would be nice if cursor: 'pointer' were set if onCellClick or onRowClick were set. Here is where it is set to 'text' in the code.
https://github.com/KevinVandy/material-react-table/blob/main/src/body/MRT_TableBodyCell.tsx#L149

### Steps to reproduce

Hovering over a table with onRowClick shows a text cursor. Selecting text fires the onRowClick behavior.

### Expected behavior

As a user, I expected a cursor: pointer to make the row click behavior more discoverable

### How often does this bug happen?

Every time

### material-react-table version

0.13.3

### Terms & Code of Conduct

- [X] I agree to follow this project's Code of Conduct
- [X] I understand that if my bug cannot be reliable reproduced in a debuggable environment, it will probably not be fixed and this issue may even be closed.

**Comment by KevinVandy** [comment]
good suggestion. Might add.

But in the meantime you can add this yourself like this

`muiTableBodyRowProps={{ sx: { cusor: 'pointer'}}}`

Also just realized `muiTableBodyRowProps` was missing from the docs props table, so will fix those docs

**Comment by KevinVandy** [comment]
Also, since you already pointed out where in the code you'd like the change, consider making the PR yourself in that conditional! Would love more code contributors.

**Comment by KevinVandy** [comment]
Will be implemented in version 14.1, authored by @ryanmk54

**Comment by KevinVandy** [comment]
@ryanmk54 Be advised that this has been reversed as I have removed all of the onClick props from use as of v0.18.0. Nothing against this PR, was just removing redundant props.

You can of course still add onClicks to table cells, and any mui components. You just have to do it like this:

```ts
<MaterialReactTable 
  muiTableBodyCellProps={{
    onClick: myOnClickFunction,
    sx: { cursor: 'pointer },
  }}
/>
```
